galley/testdatasets/validation/dataset/networking-v1alpha3-EnvoyFilter-invalid.yaml

```diff
@@ -3,5 +3,6 @@ kind: EnvoyFilter
 metadata:
   name: simple-envoy-filter
 spec:
-  workloadSelector:
-    labels:
```

Comment:
@jasonwzm with v1 CRD this is valid, before it was `The EnvoyFilter "simple-envoy-filter" is invalid: spec.workloadSelector.labels: Invalid value: "null": spec.workloadSelector.labels in body must be of type object: "null"`

Do you think its a bug its not rejected now?

Reply:
We should use `{}` for empty values. But how come this has always been passing?

Reply:
Currently, it passes with error `The EnvoyFilter "simple-envoy-filter" is invalid: spec.workloadSelector.labels: Invalid value: "null": spec.workloadSelector.labels in body must be of type object: "null"`. Error is *expected*, its a negative test case (although its broken since its supposed to check the validating webhook works, but it passes when openapi schema fails.

In CRD v1 this is now applying successfully.

Reply:
I missed the invalid part of the yaml.

Do you have a direct link of the v1 CRD used against this? I don't remember we have nullable fields for the schema.

Reply:
https://github.com/istio/istio/pull/31126

Reply:
iirc the behavior changed in k8s 1.20 so I think you need to test in that?

Reply:
I remember a similar issue